src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -1293,7 +1288,7 @@ .. chosenResolvedItem.Suppressions
                         // Suppress this dependency if PrivateAssets is set to "All"
                         if (dependency.SuppressParent == LibraryIncludeFlags.All)
                         {
-                            suppressions ??= new HashSet<LibraryDependencyIndex>();
+                            suppressions ??= new HashSet<LibraryDependencyIndex>(capacity: chosenResolvedItem.Item.Data.Dependencies.Count);
```

Comment:
Honestly, this seems really freaking large. 
If we're gonna try to be deterministic how about (chosenResolvedItem.Item.Data.Dependencies.Count-i). 

```suggestion
                            suppressions ??= new HashSet<LibraryDependencyIndex>(capacity: chosenResolvedItem.Item.Data.Dependencies.Count - i);
```

Guaranteed to avoid resizing, but with a smaller default size.